src/distributions/weighted.rs

```diff
@@ -161,10 +161,43 @@ mod test {
             assert_eq!(WeightedIndex::new(&[0, 0, 0, 0, 10, 0]).unwrap().sample(&mut r), 4);
         }
 
-        assert!(WeightedIndex::new(&[10][0..0]).is_err());
-        assert!(WeightedIndex::new(&[0]).is_err());
-        assert!(WeightedIndex::new(&[10, 20, -1, 30]).is_err());
-        assert!(WeightedIndex::new(&[-10, 20, 1, 30]).is_err());
-        assert!(WeightedIndex::new(&[-10]).is_err());
+        assert_eq!(WeightedIndex::new(&[10][0..0]).unwrap_err(), WeightedError::NoItem);
+        assert_eq!(WeightedIndex::new(&[0]).unwrap_err(), WeightedError::AllWeightsZero);
+        assert_eq!(WeightedIndex::new(&[10, 20, -1, 30]).unwrap_err(), WeightedError::NegativeWeight);
+        assert_eq!(WeightedIndex::new(&[-10, 20, 1, 30]).unwrap_err(), WeightedError::NegativeWeight);
+        assert_eq!(WeightedIndex::new(&[-10]).unwrap_err(), WeightedError::NegativeWeight);
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum WeightedError {
+    NoItem,
+    NegativeWeight,
+    AllWeightsZero,
+}
```

Comment:
Some (minimal) docs would be nice.

Reply:
I'm surprised there wasn't an error about this actually; we use `deny(missing_docs)`. But it doesn't look too bad:
https://rust-lang-nursery.github.io/rand/rand/distributions/enum.WeightedError.html

Reply:
Yes, this is weird. In any case, I opened #550 to fix this.

Reply:
The reason it wasn't an error is because we have `#[doc(hidden)]` for a bunch of stuff in `distributions/mod.rs`. They are removed in #548 though.